### DISCUSSION TITLE
Embed OsmAnd map in other application

### DISCUSSION THREAD

**Discussion by dimio** [osmandapp/OsmAnd#11241] [Q&A]
Hello!
I am new to android development. I am looking at an example of [OsmAnd-map-sample](https://github.com/osmandapp/osmand-api-demo/tree/master/OsmAnd-map-sample) and trying to re-embed an OsmAnd map in my application (like a Fragment). But I can't figure out - how can I create a `MapMarkersLayer` so that I can add my own custom markers to it? If I do not need a full-fledged OsmAnd application including a `MapActivity`. How do I get it right?
Thanks!

**Comment by Lee-Carre** [comment]
Does it *have* to be OsmAnd data format? If raster nars would be acceptable, then you might try Map(s)Forge.

Else, there is a newer vector map tile format published by Mapbox, which you might try.

**Reply by dimio** [reply]
Thanks for your help. I needed a vector map format and route building.
I have embedded part of the OsmAnd code into my application.
I'm look about Mapbox so I can use it in the future.
